source/core.py

```diff
@@ -591,6 +591,14 @@ def run(self):
 	_terminate(speech)
 	_terminate(addonHandler)
 	_terminate(garbageHandler)
+	# DMP is only started if needed.
+	# Terminate manually (and let it write to the log if necessary)
+	# as core._terminate always writes an entry.
+	try:
+		import diffHandler
+		diffHandler._dmp._terminate()
+	except Exception:
+		log.debug("Exception while terminating DMP", exc_info=True)

```

Comment:
I think this should be a higher log level, since it is only going to happen on exit it is unlikely to cause disruption for users, but it is important that we are as aware of any issues as possible. I would suggest making this an error.